.github/workflows/clp-core-build-macos.yaml

```diff
@@ -4,29 +4,27 @@ on:
   pull_request:
     paths:
       - ".github/workflows/clp-core-build-macos.yaml"
-      - "components/core/cmake/**"
-      - "components/core/CMakeLists.txt"
-      - "components/core/src/**"
-      - "components/core/tests/**"
-      - "components/core/tools/scripts/lib_install/macos/**"
-      - "components/core/tools/scripts/deps-download/**"
-      - "components/core/tools/scripts/utils/build-and-run-unit-tests.py"
+      - ".gitmodules"
```

Comment:
We should add `.gitmodules`, but otherwise I think the old set of files was more restrictive (i.e., less chance of an unrelated file like `components/core/tools/scripts/lib_install/install-boost.sh` triggering a macOS build)?

Reply:
I made the change more so to match `core-build` to be more conservative (blacklist rather than whitelist), but ya it is less restrictive. If you think it isn't worth it I can just revert (and leave in `.gitmodules`).